File: python/allergies/allergies.py
ordered = ["eggs", "peanuts", "shellfish", "strawberries", "tomatoes", "chocolate", "pollen", "cats"]
class Allergies:

    def __init__(self, score):
        self.score = score
        self.binary = [True if i == '1' else False for i in list(bin(score+256))[2::][::-1]]

    def allergic_to(self, item):
        return item in self.lst

    @property
    def lst(self):
        out = []
        for i, food in enumerate(ordered):
            if self.binary[i]:
                out.append(food)
        return out

File: ruby/allergies/allergies.rb
ALLERGIES = %w[eggs peanuts shellfish strawberries tomatoes chocolate pollen cats].freeze
class Allergies
  attr_reader :list
  def initialize(score)
    @score = score
    @list = score.to_s(2).to_i.digits.each_with_object([]).with_index { |(flag,all), idx|
      all << ALLERGIES[idx] if flag == 1 and ALLERGIES.length > idx
    }
  end
  def allergic_to?(allergen)
    @list.include?(allergen)
  end
end
